src/cmd-handlers.ts

```diff
@@ -1,65 +1,72 @@
-function make_actions(reqFn: CallableFunction, body_args: Array<string> = [], include?: string[]) {
+import { IncludeFromEnum, Entity, FieldModify, ActionDetails } from "./types"
+
+function make_actions(reqFn: CallableFunction, action_details: ActionDetails) {
   async function load(this:any, msg: any) {
-    const source = {...msg.q}
-    let body: Record<string,any> = {}
+    const args = {...msg.q}
+    const { modify } = action_details
 
-    const old_source = {...source}
+    const old_args = {...args}
 
-    if(source.repo_id) {
-      body = owner_repo(source.repo_id)
-      delete source.repo_id
-    }
+    let body = basic_body(args)
+
+    const res = await reqFn(body)
+    
+    let entity: Entity = this.make$(msg.ent.entity$).data$(res.data)
 
-    let res = await reqFn({...body, ...source})
-    res = res.data
-
-    if (include) {
-      include.forEach((item) => {
-        if (item.indexOf(' as ') !== -1) {
-          const [attr, new_attr_name] = item
-            .split(' as ')
-            .map((item) => item.trim())
-          res[new_attr_name] = old_source[attr]
-        } else {
-          res[item] = old_source[item]
-        }
+    if(modify) {
+      const replacements = modify.filter(mod => mod.replace_for !== undefined)
+      entity = ent_replacements(entity, replacements, {
+        args: old_args,
+        res
       })
+      
+      const renamings = modify.filter(mod => mod.rename !== undefined)
+      entity = ent_renamings(entity, renamings)
     }
 
-    return this.make$(msg.ent.entity$).data$(res)
+    return entity
   }
 
   async function save(this:any, msg: any) {
-    const source = {...msg.ent}
+    const entity = {...msg.ent}
```

Comment:
Is this necessary? Would this work:
```
const ent = msg.ent
const q = msg.q
```

I don't see where these objects are modified, so created new vars seems unnecessary?

Reply:
Richard, I used destructuring here : 
`let body = basic_body({...msg.q})` 

because I delete repo_id in order to create a "basic" body before calling the request callback, as per 
```
  function basic_body(source: Record<string, any>) {
    let body: Record<string,any> = {}

    if(source.repo_id) {
      body = owner_repo(source.repo_id)
    }

    delete source.repo_id

    return {...body, ...source}
  }
```
If I didn't do like that, the arguments passed by the plugins users would ended up being modified, in case they were passed by reference. It happened just now when running the tests, for exemple.

Reply:
Richard, I used destructuring here : 
`let body = basic_body({...msg.q})` 

because repo_id is deleted in order to create a "basic" body before calling the request callback, as per 
```
  function basic_body(source: Record<string, any>) {
    let body: Record<string,any> = {}

    if(source.repo_id) {
      body = owner_repo(source.repo_id)
    }

    delete source.repo_id

    return {...body, ...source}
  }
```
If I didn't do like that, the arguments passed by the plugins users would ended up being modified, in case they were passed by reference. It happened just now when running the tests, for exemple. So in this case, if I call entity.load$(reference) two times, it won't work without `{...msg.q}` .